File: chat-uikit/src/main/java/io/agora/chat/uikit/widget/chatrow/AutolinkSpan.java
package io.agora.chat.uikit.widget.chatrow;

import android.text.style.URLSpan;
import android.view.View;

import io.agora.chat.uikit.R;


public class AutolinkSpan extends URLSpan {

    public AutolinkSpan(String url) {
        super(url);
    }

    @Override
    public void onClick(View widget) {
        if (widget.getTag(R.id.action_chat_long_click) != null) {
            widget.setTag(R.id.action_chat_long_click, null);
            return;
        }
        super.onClick(widget);
    }
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/widget/chatrow/AutolinkSpan.kt
package com.hyphenate.easeui.widget.chatrow

import android.text.style.URLSpan
import android.view.View
import com.hyphenate.easeui.R

class AutolinkSpan(url: String?) : URLSpan(url) {
    override fun onClick(widget: View) {
        if (widget.getTag(R.id.action_chat_long_click) != null) {
            widget.setTag(R.id.action_chat_long_click, null)
            return
        }
        super.onClick(widget)
    }
}